solidity/random-beacon/deploy/04_deploy_random_beacon.ts

```diff
@@ -0,0 +1,83 @@
+import type { HardhatRuntimeEnvironment } from "hardhat/types"
+import type { DeployFunction } from "hardhat-deploy/types"
+
+const func: DeployFunction = async (hre: HardhatRuntimeEnvironment) => {
+  const { getNamedAccounts, deployments, helpers } = hre
+  const { deployer } = await getNamedAccounts()
+
+  const T = await deployments.get("T")
+  const TokenStaking = await deployments.get("TokenStaking")
+  const ReimbursementPool = await deployments.get("ReimbursementPool")
+  const BeaconSortitionPool = await deployments.get("BeaconSortitionPool")
+
+  const BLS = await deployments.deploy("BLS", {
+    from: deployer,
+    log: true,
+  })
+
+  const BeaconAuthorization = await deployments.deploy("BeaconAuthorization", {
+    from: deployer,
+    log: true,
+  })
+
+  const BeaconDkg = await deployments.deploy("BeaconDkg", {
+    from: deployer,
+    log: true,
+  })
+
+  const BeaconInactivity = await deployments.deploy("BeaconInactivity", {
+    from: deployer,
+    log: true,
+  })
+
+  const BeaconDkgValidator = await deployments.deploy("BeaconDkgValidator", {
+    from: deployer,
+    args: [BeaconSortitionPool.address],
+    log: true,
+  })
+
+  const RandomBeacon = await deployments.deploy("RandomBeacon", {
+    contract:
```

Comment:
What is the `contract` option doing and what is happening when it's undefined? Can you provide a reference to the documentation where it is described?

Reply:
My understanding is that it is used for naming purposes, ie `contract:RandomBeaconStub` this is what is being actually deployed, but we can refer to it in the rest of the code as `RandomBeacon`. Similar to what we have for the sortition pools. We deploy the original `SortitionPool` but then we can refer to it as a `BeaconSortitionPool` or `EcdsaSortitionPool`. Is this right, @nkuba?

I wasn't able to find the docs for this `contract` option, but the [source code](https://github.com/wighawag/hardhat-deploy/blob/master/src/helpers.ts#L495) tells me that instead of `undefined` we should probably have `RandomBeacon` in case the `network !== hardhat`. @nkuba can you pls confirm?

Reply:
> My understanding is that it is used for naming purposes, ie `contract:RandomBeaconStub` this is what is being actually deployed for hardhat, but we can refer to it in the rest of the code as `RandomBeacon`. Similar to what we have for the sortition pools. We deploy the original `SortitionPool` but then we can refer to it as a `BeaconSortitionPool` or `EcdsaSortitionPool` depending on the "project". Is this right, @nkuba?

:100:

> I wasn't able to find the docs for this `contract` option, but the [source code](https://github.com/wighawag/hardhat-deploy/blob/master/src/helpers.ts#L495) tells me that instead of `undefined` we should probably have `RandomBeacon` in case the `network !== hardhat`. @nkuba can you pls confirm?

I'd leave it as `undefined` as it will default to `RandomBeacon` anyways then and in case we copy-paste the code we have one line less to change. According to docs:
```
contract?: // this is an optional field. If not specified it defaults to the contract with the same name as the first parameter
```